elliptic-curve/src/hash2field/expand_msg_xmd.rs

```diff
@@ -1,17 +1,20 @@
+use core::ops::Mul;
+
 use super::{Domain, ExpandMsg};
 use digest::{
     generic_array::{
-        typenum::{IsLess, IsLessOrEqual, Unsigned, U256},
+        typenum::{IsLess, IsLessOrEqual, Unsigned, U255, U256, U65536},
         GenericArray,
     },
     BlockInput, Digest,
 };
+use generic_array::{typenum::Prod, ArrayLength};
 
 /// Placeholder type for implementing expand_message_xmd based on a hash function
 pub struct ExpandMsgXmd<HashT>
 where
     HashT: Digest + BlockInput,
-    HashT::OutputSize: IsLess<U256>,
+    HashT::OutputSize: IsLessOrEqual<U256>,
```

Comment:
This should have been `<=256`, not `<256` as far as I can tell from the spec?

Reply:
I originally suggested `<256` so the cast to `u8` wouldn't overflow.

I'm a bit confused what's even happening now as you appear to be passing `[usize]` to `chain`?!

Reply:
No, I messed up, it should be less then 256, I was a lunatic and thought `u8` can reach 256.

Reply:
Fixed.